### ISSUE TITLE
Querying from startup_items table will cause mounting all users home dirs from /home, if automounts are configured

### ISSUE THREAD

**Issue by alexbozhenko** [osquery/osquery#6796]
# Bug report

If autofs configured to mount `/home` directories from nfs, querying `startup_items` table will cause mounting all users home dirs.
That means querying from this table on all boxes in the company(e.g. using `fleet`) will mount ALL users home directoriess on ALL boxes at the same time.
NFS server may be not happy about that.

### What operating system and version are you using?
`CentOS Linux release 7.9.2009`

### What version of osquery are you using?
`4.5.1`

### What steps did you take to reproduce the issue?

```
# mount | wc -l
57
# osqueryi --line "select * from startup_items;" >& /dev/null
# mount | wc -l
1537
```

### What did you expect to see?
Well, since auto-mounts are configured, I think there is nothing to fix here on osquery side? 
But the situation may be not ideal, so I just wanted to share it here, so others are aware.

**Comment by theopolis** [comment]
We do try to handle this situation in osquery: https://github.com/osquery/osquery/blob/4cf19f29c2508cc5821b82017029da0ada16f9e1/osquery/filesystem/linux/mounts.cpp#L78 

But `startup_items` uses a third party systemd/dbus library to scan contents. I think we should have a closer look to see if we can disable the unwanted auto-mounting.

**Comment by Smjert** [comment]
I don't think this is caused by dbus, but the fact that we access the home directories to find autostart scripts.

This is a bit tricky because if we wanted to avoid this, I'm not sure there's a way to verify that a certain path is really present without triggering the mount.

**Comment by randalmichael** [comment]
I have disabled all of the "mounts" and "startup_items" queries relating to Linux systems in Fleet and I still have the same issue with all of the user home directories NFS mounting at once. What other queries could cause this behavior? The OSquery clients do not have any queries configured in the local configuration. All queries are scheduled from Fleet. This problem is causing the "automount" process to consume 60%+ of the CPU.
